src/aliceVision/sfmData/sfmData_test.cpp

```diff
@@ -0,0 +1,52 @@
+
+#include <boost/filesystem.hpp>
+#include <aliceVision/sfmData/SfMData.hpp>
+
+#define BOOST_TEST_MODULE sfmData
+#include <boost/test/included/unit_test.hpp>
+
+using namespace aliceVision;
+namespace fs = boost::filesystem;
+
+BOOST_AUTO_TEST_CASE(SfMData_InternalFolders)
+{
+  const std::string filename = "InternalFolders.sfm";
+  sfmData::SfMData sfmData;
+
+  // add relative features/matches folders with duplicates
+  std::string refFolder("..");
+  sfmData.addFeaturesFolders({refFolder, refFolder});
+  sfmData.addMatchesFolders({refFolder, refFolder});
+  auto featuresFolders = sfmData.getFeaturesFolders();
+  auto matchesFolders = sfmData.getMatchesFolders();
+  // ensure duplicates were removed
+  BOOST_CHECK_EQUAL(featuresFolders.size(), 1);
+  BOOST_CHECK_EQUAL(matchesFolders.size(), 1);
+  // sfmData has no absolute path set, folders are still in relative form
+  BOOST_CHECK_EQUAL(featuresFolders[0], refFolder);
+  BOOST_CHECK_EQUAL(matchesFolders[0], refFolder);
+
+  // set absolutePath to current filename
+  sfmData.setAbsolutePath(fs::absolute(filename).string());
+  featuresFolders = sfmData.getFeaturesFolders();
+  matchesFolders = sfmData.getMatchesFolders();
+  // internal folders were kept...
+  BOOST_CHECK_EQUAL(featuresFolders.size(), 1);
+  BOOST_CHECK_EQUAL(matchesFolders.size(), 1);
+  // ... and are now absolute paths
+  BOOST_CHECK(fs::path(featuresFolders[0]).is_absolute());
+  BOOST_CHECK(fs::equivalent(featuresFolders[0], refFolder));
+  BOOST_CHECK(fs::path(matchesFolders[0]).is_absolute());
+  BOOST_CHECK(fs::equivalent(matchesFolders[0], refFolder));
```

Comment:
We could try to add a new folder, like:
```
sfmData.setAbsolutePath(fs::absolute(filename).string());
sfmData.addFeaturesFolder(fs::absolute(filename) / "uselessFolder/..").string()); // duplicates from non-canonical path
```
to check the canonical path conversion with duplicates from different input strings.